modules/custom-status/custom-status.php

```diff
@@ -281,55 +227,60 @@ public function action_admin_enqueue_scripts() {
 		}
 	}
 
-	public function load_scripts_for_block_editor() {
+	/**
+	 * Enqueue resources that we need in the admin settings page
+	 *
+	 * @access private
+	 */
+	public static function load_scripts_for_block_editor(): void {
 		$asset_file = include VIP_WORKFLOW_ROOT . '/dist/modules/custom-status/custom-status-block.asset.php';
 		wp_enqueue_script( 'vip-workflow-block-custom-status-script', VIP_WORKFLOW_URL . 'dist/modules/custom-status/custom-status-block.js', $asset_file['dependencies'], $asset_file['version'], true );
 
-		$publish_guard_enabled = ( 'on' === VIP_Workflow::instance()->settings->module->options->publish_guard ) ? true : false;
-
-		$custom_statuses = $this->get_custom_statuses_with_editorial_metadata();
+		$publish_guard_enabled = ( 'on' === OptionsUtilities::get_options_by_key( 'publish_guard' ) ) ? true : false;
 
 		wp_localize_script( 'vip-workflow-block-custom-status-script', 'VW_CUSTOM_STATUSES', [
 			'current_user_id'          => get_current_user_id(),
 			'is_publish_guard_enabled' => $publish_guard_enabled,
-			'status_terms'             => $custom_statuses,
-			'supported_post_types'     => VIP_Workflow::instance()->get_supported_post_types(),
+			'status_terms'             => self::modify_custom_statuses_with_editorial_metadata(),
+			'supported_post_types'     => HelperUtilities::get_supported_post_types(),
 		] );
 	}
 
 	/**
-	 * Get the custom statuses, with the editorial metadatas populated under meta.
+	 * Modify the custom statuses to include the editorial metadatas for UI purposes.
 	 *
-	 * This function is only necessary until the big refactor is done. After that, it should not be necessary.
+	 * This isn't done anywhere else due to the taxonomies being registered at different times.
+	 * In addition, registering the taxonomies in the wrong order can cause the manage posts page to break
+	 * as well as the default status for a post itself.
 	 *
-	 * @return array $custom_statuses The custom statuses with editorial metadatas
+	 * @return array $custom_statuses The custom statuses with the editorial metadatas included
 	 */
-	private function get_custom_statuses_with_editorial_metadata() {
+	private static function modify_custom_statuses_with_editorial_metadata(): array {
```

Comment:
I made a much more optimized version of the previous method. I also updated the description to mention why it's there. No matter what I did, something broke so I had to do it this way.

If there is a good way to link two taxonomies together to mirror how a foreign key relationship works then we can do that. Otherwise, we need to add this in. 

IMO, I'm debating if it's worth just taking this out and instead doing this in another PR:

- Add a get and get by slug/id Rest api for CS and EM
- Scrap the UI specific method and put it all together within JS itself using the above rest endpoints
- That would do the trick as this field is only for UI purposes anyways

Reply:
This is still 👍 for now, just thinking about how we'd change this in the future.